api/provider/aws/test_aws/load_balancer_update_test.go

```diff
@@ -0,0 +1,149 @@
+		models.Port{
+			CertificateName: "cert",
+			ContainerPort:   8080,
+			HostPort:        8088,
+			Protocol:        "http",
+		},
+		models.Port{
+			CertificateName: "cert",
+			ContainerPort:   4444,
+			HostPort:        444,
+			Protocol:        "https",
+		},
+	}
+
+	requestHealthCheck := &models.HealthCheck{
+		Target:             "HTTPS:444/path/to/site",
+		Interval:           15,
+		Timeout:            10,
+		HealthyThreshold:   5,
+		UnhealthyThreshold: 4,
+	}
+
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: "lb_name",
+		Ports:          &requestPorts,
+		HealthCheck:    requestHealthCheck,
+	}
+
+	configureHealthCheckInput := &elb.ConfigureHealthCheckInput{}
+	configureHealthCheckInput.SetLoadBalancerName("l0-test-lb_name")
+	configureHealthCheckInput.SetHealthCheck(healthCheckHelper(requestHealthCheck))
+
+	configureHealthCheckOutput := &elb.ConfigureHealthCheckOutput{}
+	configureHealthCheckOutput.SetHealthCheck(healthCheckHelper(requestHealthCheck))
+
+	mockAWS.ELB.EXPECT().
+		ConfigureHealthCheck(configureHealthCheckInput).
+		Return(configureHealthCheckOutput, nil)
+
+	certificateARN := "arn:aws:iam::123456789012:server-certificate/cert"
+	serverCertificateMetadata := &iam.ServerCertificateMetadata{}
+	serverCertificateMetadata.SetArn(certificateARN)
+	serverCertificateMetadata.SetServerCertificateName("cert")
+	serverCertificateMetadataList := []*iam.ServerCertificateMetadata{serverCertificateMetadata}
+
+	listServerCertificatesOutput := &iam.ListServerCertificatesOutput{}
+	listServerCertificatesOutput.SetServerCertificateMetadataList(serverCertificateMetadataList)
+
+	mockAWS.IAM.EXPECT().
+		ListServerCertificates(&iam.ListServerCertificatesInput{}).
+		Return(listServerCertificatesOutput, nil).
+		AnyTimes()
+
+	readSGHelper(mockAWS, "l0-test-lb_name-lb", "lb_sg")
+	listenerDescription := &elb.ListenerDescription{}
+	listenerDescription.SetListener(listenerHelper(provider.DefaultLoadBalancerPort))
+
+	lb := &elb.LoadBalancerDescription{}
+	lb.SetLoadBalancerName("l0-test-lb_name")
+	lb.SetHealthCheck(healthCheckHelper(&provider.DefaultHealthCheck))
+	lb.SetListenerDescriptions([]*elb.ListenerDescription{listenerDescription})
+
+	describeLoadBalancersInput := &elb.DescribeLoadBalancersInput{}
+	describeLoadBalancersInput.SetLoadBalancerNames([]*string{aws.String("l0-test-lb_name")})
+	describeLoadBalancersInput.SetPageSize(1)
+
+	describeLoadBalancersOutput := &elb.DescribeLoadBalancersOutput{}
+	describeLoadBalancersOutput.SetLoadBalancerDescriptions([]*elb.LoadBalancerDescription{lb})
+
+	mockAWS.ELB.EXPECT().
+		DescribeLoadBalancers(describeLoadBalancersInput).
+		Return(describeLoadBalancersOutput, nil)
+
+	revokeIngressInput := revokeSGIngressHelper(provider.DefaultLoadBalancerPort)
+	revokeIngressInput.SetGroupId("lb_sg")
+
+	mockAWS.EC2.EXPECT().
+		RevokeSecurityGroupIngress(revokeIngressInput).
+		Return(&ec2.RevokeSecurityGroupIngressOutput{}, nil)
+
+	port := int64(80)
```

Comment:
this should probably reference the default port for clarity, e.g. `DefaultLBPort.HostPort`

Reply:
Ah, good point. :white_check_mark: